src/analysis/lattice.h

```diff
@@ -0,0 +1,53 @@
+#ifndef wasm_analysis_lattice_h
+#define wasm_analysis_lattice_h
+
+#include "wasm.h"
+#include <bitset>
+#include <iostream>
+
+namespace wasm::analysis {
+
+enum LatticeComparison { NO_RELATION, EQUAL, LESS, GREATER };
+
+// Represents a powerset lattice element (i.e. a set) as a bitvector. A true
+// means that an element is present in the set.
+template<size_t N> struct BitsetPowersetLattice {
+  std::bitset<N> value;
+
+  // Copies sets the bitvector to be identical to that of a source lattice
+  // element.
+  inline void copy(BitsetPowersetLattice<N>& src);
```

Comment:
Removed the `inline`s in the header files, but not the `impl` header files. The reason mentioned in our discussion was that `inline` is implicitly applied to all member methods inside the scope of a header file's class definition. We need inlines in the `impl` header files because these function definitions are not within the class definition scope.